api/ruby/ghe-org-permissions-report.rb

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env ruby
+# Generates a CSV report listing all organizations, their repositories,
+# collaborators, effective permissions, teams, and team permissions
+#
+# Set OCTOKIT_ACCESS_TOKEN to a token with read:org scope owned by a site admin
+# and OCTOKIT_API_ENDPOINT to http(s)://[your-hostname]/api/v3/
+#
+# Use `ghe-org-admin-promote` to make a site admin an owner of all
+# organizations
+require 'octokit'
+
+Octokit.default_media_type = 'application/vnd.github.ironman-preview+json'
+ghe = Octokit::Client.new
+
+PERMISSION_LEVELS = [:admin, :push, :pull]
+
+def get_repo_teams(ghe, repo_full_name)
+  teams = []
+  ghe.repo_teams(repo_full_name).each do |t|
+    teams << [t, ghe.team_members(t.id).map(&:login)]
+  end
+
+  teams
+rescue Octokit::NotFound
+  []
+end
+
+def get_org_role(ghe, org_name, user_login)
+  ghe.org_membership(org_name, user: user_login).role
+rescue Octokit::NotFound
+  'outside-collaborator'
+end
+
+permission_list = []
+ghe.orgs(ghe.user).each do |org|
+  # We shouldn't try to get the org permissions if we're not an admin,
+  # they'll be wrong or misleading
+  if get_org_role(ghe, org.login, ghe.user.login) != 'admin'
+    puts "Skipping #{org.login} - not an organization admin"
```

Comment:
❤️ this functionality. I'll need to add it to our older scripts where relevant.

Perhaps we should change it to use `STDERR.puts`, that way if the user is redirecting STDOUT to a .csv file, they won't get the error messages in there?